apps/frontend-pwa/src/pages/index.tsx

```diff
@@ -147,62 +149,119 @@ const Index = function () {
     return <div>loading...</div>
   }
 
-  async function onSubscribeClick(subscribed: boolean, courseId: string) {
+  async function onSubscribeClick(
```

Comment:
we should think about moving these generic hooks to shared components, as we might implement push on other PWA as well. also, would make this component much cleaner. the mutation to actually execute could be passed in as a param to the hook to make it generic.